pkg/reconciler/openshift/tektonaddon/communityClusterTasks.go

```diff
@@ -0,0 +1,146 @@
+)
+
+var communityResourceURLs = []string{
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/jib-maven/0.4/jib-maven.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/helm-upgrade-from-source/0.3/helm-upgrade-from-source.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/helm-upgrade-from-repo/0.2/helm-upgrade-from-repo.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/git-cli/0.3/git-cli.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/pull-request/0.1/pull-request.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/master/task/kubeconfig-creator/0.1/kubeconfig-creator.yaml",
+	"https://raw.githubusercontent.com/tektoncd/catalog/main/task/argocd-task-sync-and-wait/0.2/argocd-task-sync-and-wait.yaml",
+}
+
+func (r *Reconciler) EnsureCommunityClusterTask(ctx context.Context, enable string, ta *v1alpha1.TektonAddon) error {
+
+	communityClusterTaskLabelSelector, err := common.LabelSelector(communityClusterTaskLS)
+	if err != nil {
+		return err
+	}
+
+	if enable == "true" {
+
+		exist, err := checkIfInstallerSetExist(ctx, r.operatorClientSet, r.operatorVersion, communityClusterTaskLabelSelector)
+		if err != nil {
+			return err
+		}
+
+		if !exist {
+			msg := fmt.Sprintf("%s being created/upgraded", CommunityClusterTaskInstallerSet)
+			ta.Status.MarkInstallerSetNotReady(msg)
+			return r.ensureCommunityClusterTasks(ctx, ta)
+		}
+
+		if err := r.checkComponentStatus(ctx, communityClusterTaskLabelSelector); err != nil {
+			ta.Status.MarkInstallerSetNotReady(err.Error())
+			return nil
+		}
+
+	} else {
+		// if disabled then delete the installer Set if exist
+		if err := r.deleteInstallerSet(ctx, communityClusterTaskLabelSelector); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// installerset for communityclustertask
+func (r *Reconciler) ensureCommunityClusterTasks(ctx context.Context, ta *v1alpha1.TektonAddon) error {
+
+	if SkipCommunityTaskFetch(retryWaitTime) {
+		return nil
+	}
+
+	communityClusterTaskManifest := mf.Manifest{}
+	if err := r.appendCommunityTarget(ctx, &communityClusterTaskManifest, ta); err != nil {
+		retryWaitTime = time.Now().Add(15 * time.Minute).Format(layout)
+
+		// Continue if failed to resolve community task URL.
+		// (Ex: on disconnected cluster community tasks won't be reachable because of proxy).
+		logging.FromContext(ctx).Error("Failed to get community task: Skipping community tasks installation  ", err)
+		r.enqueueAfter(r, 10*time.Second)
+		return nil
+	} else {
+		if err := r.communityTransform(ctx, &communityClusterTaskManifest, ta); err != nil {
+			return err
+		}
+
+		communityClusterTaskManifest = communityClusterTaskManifest.Append(communityClusterTaskManifest)
+	}
+	if err := createInstallerSet(ctx, r.operatorClientSet, ta, communityClusterTaskManifest,
```

Comment:
```suggestion
	if err := r.appendCommunityTarget(ctx, &communityClusterTaskManifest, ta); err != nil {
		retryWaitTime = time.Now().Add(15 * time.Minute).Format(layout)

		// Continue if failed to resolve community task URL.
		// (Ex: on disconnected cluster community tasks won't be reachable because of proxy).
		logging.FromContext(ctx).Error("Failed to get community task: Skipping community tasks installation  ", err)
		r.enqueueAfter(r, 10*time.Second)
		return nil
	}
	if err := r.communityTransform(ctx, &communityClusterTaskManifest, ta); err != nil {
		return err
	}
	communityClusterTaskManifest = communityClusterTaskManifest.Append(communityClusterTaskManifest)

	if err := createInstallerSet(ctx, r.operatorClientSet, ta, communityClusterTaskManifest,
```

the else is redundant, there is no other code path.
this function is basically 3 error checks.

Reply:
that else is like saying
```golang
if err != nil {
     return err
} else {
   next action
} 
```

instead the code flow could be
```golang
if err != nil {
     return err
}
next action
```